packages/block-library/src/navigation/index.php

```diff
@@ -148,10 +161,29 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 		)
 	);
 
+	$modal_unique_id = uniqid();
+
+	// Determine whether or not navigation elements should be wrapped in the markup required to make it responsive.
+	$responsive_container_markup = isset( $attributes['responsiveNavigation'] ) && true === $attributes['responsiveNavigation'] ? sprintf(
+		'<button class="wp-block-navigation__responsive-container-open" data-micromodal-trigger="modal-%2$s" aria-label="Open menu"><svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" focusable="false"><rect x="4" y="7.5" width="16" height="1.5" /><rect x="4" y="15" width="16" height="1.5" /></svg></button>
```

Comment:
The button misses some a11y properties, see Reakit:

<img width="994" alt="Screen Shot 2021-04-16 at 10 31 46" src="https://user-images.githubusercontent.com/699132/114996316-0113de00-9e9f-11eb-9e28-48174ae3b880.png">

It looks like we should add `aria-haspopup` and `aria-expanded`. @diegohaz can you confirm?

Reply:
Those attributes are not required for modal triggers. But they help a lot as screen readers will announce the button as a popup button and whether it's expanded or not.